src/packet_analysis/protocol/udp/UDP.cc

```diff
@@ -223,11 +223,17 @@ void UDPAnalyzer::DeliverPacket(Connection* c, double t, bool is_orig, int remai
 	// port here because the orig/resp should have already swapped around based on
 	// likely_server_ports. This also prevents us from processing things twice if protocol
 	// detection has to be used.
-	ForwardPacket(std::min(len, remaining), data, pkt, ntohs(c->RespPort()));
+	ForwardPacket(len, data, pkt, ntohs(c->RespPort()));
 
 	// Also try sending it into session analysis.
 	if ( remaining >= len )
+		{
+		// Adjust the remaining by the difference between len and caplen in the packet, since
+		// we might not actually have all of the data available.
```

Comment:
But at the same time this means `remaining` was already out of sync with how much is left in `data` which seems very dangerous. It would be much easier if `remaining` is always guaranteed to be `<= bytes left in data`, or else seems this special case would also be required in TCP in case there were TCP in TCP tunnels?

Reply:
It's probably the latter, that TCP also needs the change. It only came up in UDP here because that's the test cases we have.